flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/types/InternalSerializers.java

```diff
@@ -71,10 +75,12 @@ public static TypeSerializer create(LogicalType type, ExecutionConfig config) {
 				DecimalType decimalType = (DecimalType) type;
 				return new DecimalSerializer(decimalType.getPrecision(), decimalType.getScale());
 			case ARRAY:
-				return BinaryArraySerializer.INSTANCE;
+				return new BaseArraySerializer(((ArrayType) type).getElementType(), config);
```

Comment:
Each time requires creating a new serializer. Does it cause performance degrade?

Reply:
InternalSerializers.create is used to create serializers. runtime per record process should not invoke it.